### ISSUE TITLE
Crashing on Windows 10

### ISSUE THREAD

**Issue by flip4bytes** [winnfsd/vagrant-winnfsd#65]
Using this with Homestead on Windows 10 and I keep getting "winnfsd.exe has stopped working" - It seems to work great for the initial load of the site sometimes, but immediately afterwards crashes and I have to halt and then up the server again. Makes using winnfsd unusable on Windows 10.

**Comment by worthwhileindustries** [comment]
This is definitely a blocker to my upgrade to Windows 10

**Comment by flip4bytes** [comment]
It seems to be a bit random. I use OctoberCMS on some sites and Laravel on other sites. OctoberCMS runs on Laravel, so it shouldn't be much different? But it crashes when on my client's OctoberCMS sites, and seems to run alright for the most part on just barebones Laravel sites. Though I haven't used it enough to know if it crashes too on the laravel sites. Just messing around a bit it seems to work alright. Can't use winnfsd though seeing as most of my client's sites are running on October.

**Comment by worthwhileindustries** [comment]
I've had so many issues with the actual winnfsd.exe ... with recently seeming to be pretty stable after the latest releases. I'm running Win 8.1... with Vagrant 1.74, and Virtualbox 5. Have you tried updating all the things. Also, I agree it should be the same way barebones or not. However, sometimes i get a stack trace running vagrant commands and the second or third time they work. Vagrant == random.

**Comment by flip4bytes** [comment]
I too have Vagrant 1.74, and Virtualbox 5 - Just updated today. Have been trying to get my workflow up and running again since updating to Windows 10. That's what I have spent all day today doing. But yeah, things definitely seem a bit random unfortunately. Really hope to get this up and working because not having nsf turned on makes things sooooooo slow, I can't take it haha

**Comment by worthwhileindustries** [comment]
Yep I feel about a 4x increase in speed for sure. Shame MS doesn't support this natively.

**Comment by flip4bytes** [comment]
There's so many little things that make dev such a pain on Windows.. But I've always been a windows user and Windows 10 is honestly phenominal. They have done an excellent job with it.

**Comment by worthwhileindustries** [comment]
Ha yeah I agree but, I try to be platform agnostic professionally. On personal level though... yeah they've done a great job. With access to VM's it's the best of both worlds.

**Comment by yannschepens** [comment]
Hi guy's,

Thank for returns, i haven't Windows 10 yet. But you can check if Windows
10 support nfs natively. If yes, we just gonna have to remove winnfsd and
try to connect directly with Windows nfs server. If not we have some hard
work to do I think ;-)

**Comment by worthwhileindustries** [comment]
@yannschepens I'm 99% certain Microsoft has no intentions of supporting NFS natively in Windows.

**Comment by yannschepens** [comment]
Hum samba....

**Comment by worthwhileindustries** [comment]
Yeah samba can be good but, doesn't even come close to the speed of this nfs plugin.

**Comment by flip4bytes** [comment]
@yannschepens Thanks for the reply. I looked a bit into it, and like @worthwhileindustries mentioned, Windows has absolutely no plans of bringing NFS to Windows and there's nothing that points in the direction that they are going to change that, so I don't think winnfsd is going anywhere.

Thanks for looking into this!

**Comment by loostro** [comment]
Is winnfsd going to support Windows 10?

**Comment by integratedforpublishers** [comment]
We use vagrant-winnfsd here on a several PC's with Windows 10. We don't have problems with it.

**Comment by loostro** [comment]
It seems my configuration was incorrect. Another plugin `vagrant-cachier` as descibed [here](http://jeremybarthe.com/2015/02/02/speed-up-vagrant-environment-symfony2/) is not working with winfsd (unknown `ver=3` and other mount options).

Winfsd by itself seems to work, but... it just crashed during composer install :P

**Comment by skonrad** [comment]
As this issue is still open and I'am able to reproduce it perfectly I have some notes that might be useful.

I just installed 1.2.0 and 1.2.1 with the service crashing.
The only difference in 1.2.1 is, that it's crashing in the background and windows is no longer telling you that the service crashed.

After downgrading to 1.1.0 I can no longer reproduce this behaviour.

Tested with win10 on the latest updates.
PS: I set the service (winnfsd.exe) to run as administrator with the option set at the compatibility-tab to allow symlinks within the shared directory.

**Comment by marcharding** [comment]
Can you try if the exe linked in this comment https://github.com/winnfsd/winnfsd/issues/22#issuecomment-232504404 solves this?

**Comment by skonrad** [comment]
> Can you try if the exe linked in this comment winnfsd/winnfsd#22 (comment) solves this?

Will do so as soon as I got some spare time.

**Comment by jamesmacwhite** [comment]
@skonrad Running `vagrant up` from an elevated CMD prompt should be enough for the administrator stuff, you shouldn't need to edit the executable itself. I certainly don't have to on Windows 10.

**Comment by cbj4074** [comment]
For my own edification, why is it necessary to run `vagrant up` from an elevated command-prompt? Is this a requirement of Vagrant? Or of this plugin?

I'm able to use this plugin perfectly well without having to elevate. I do have crashing issues, but they don't seem to be related.

Or is elevation necessary to enable symlinking in Windows? I ask mostly because there's a means by which to enable non-Administrators to create symlinks on Windows via Group Policy.

Thanks for any elaboration in this regard!

**Comment by jamesmacwhite** [comment]
@cbj4074 it is for allowing symlinks when using commands like npm

**Comment by cbj4074** [comment]
@jamesmacwhite Thanks for clarifying!

Just throwing out another option for peoples' consideration, in the event that elevation is not an option or the operator would prefer to avoid it:

http://superuser.com/questions/104845/permission-to-make-symbolic-links-in-windows-7
